lib/vm.py

```diff
@@ -361,6 +365,26 @@ def test_snapshot_on_running_vm(self):
         finally:
             snapshot.destroy(verify=True)
 
+    def vdi_resize(self, vdi_uuid, new_size=None):
+        # Reference
+        MAX_VHD_SIZE = 2088960 * 1024 * 1024 # 2088960 MB
+        MIN_VHD_SIZE = 1 * 1024 * 1024 # 1MB
+
+        if self.is_running():
+            self.shutdown(verify=True, force_if_fails=True)
```

Comment:
I'm not sure about that. We've been tempted in the past to do smart things like that but we regretted it and came back to simple things, such as asserts. I think we should simply not call vdi_resize on a running VM at the moment, or, when we do, let it be so that we check that we get an appropriate error message.

Later, when we actually support resizing VDIs without turning the VMs off, we'd have to change this anyway.

Reply:
Right, will update it.